client/beefy/src/worker.rs

```diff
@@ -430,6 +444,20 @@ where
 	}
 }
 
+/// Returns a future that waits until major sync is done
+/// before completing
+fn wait_for_major_syncing<SO: SyncOracle + Send + Sync + Clone + 'static>(
+	mut sync_oracle: SO,
+) -> impl future::Future<Output = ()> {
+	return future::poll_fn(move |cx| {
+		if sync_oracle.is_major_syncing() {
+			cx.waker().wake_by_ref();
```

Comment:
this is equivalent to a busy-wait loop, right? Like @seunlanlege said, this future needs to be tied to the sync_oracle waker.